packages/dds/tree/src/feature-libraries/forest-summary/incrementalSummaryBuilder.ts

```diff
@@ -0,0 +1,430 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { assert } from "@fluidframework/core-utils/internal";
+import type { IExperimentalIncrementalSummaryContext } from "@fluidframework/runtime-definitions/internal";
+import {
+	SummaryTreeBuilder,
+	type ReadAndParseBlob,
+} from "@fluidframework/runtime-utils/internal";
+import {
+	brand,
+	setInNestedMap,
+	tryGetFromNestedMap,
+	type NestedMap,
+} from "../../util/index.js";
+import type {
+	ChunkReferenceId,
+	EncodedFieldBatch,
+	IncrementalEncoderDecoder,
+	TreeChunk,
+} from "../chunked-forest/index.js";
+import type {
+	FieldKey,
+	ITreeCursorSynchronous,
+	TreeNodeSchemaIdentifier,
+} from "../../core/index.js";
+import { SummaryType } from "@fluidframework/driver-definitions";
+import type { IChannelStorageService } from "@fluidframework/datastore-definitions/internal";
+import type { ISnapshotTree } from "@fluidframework/driver-definitions/internal";
+import { LoggingError } from "@fluidframework/telemetry-utils/internal";
+
+/**
+ * The key for the summary blob that contains the contents of an incremental chunk.
```

Comment:
I don't find this a good explanation. Presumably there is some summary tree node for the incremental chunk, and this key is for the blob under it, which contains the "inline" portion of that chunk encoded with the forest codec. That however is just a portion of the data: calling it the "contents of an incremental chunk" makes it sound like there is no other content under the summary tree node for the chunk, which if that was the case would make having both summary nodes pointless and give us no place to put child summary tree nodes).

Also somewhere there should be documentation explain the whole format, and this should probably reference that.